arthur/tasks.py

```diff
@@ -204,13 +208,16 @@ def remove(self, task_id):
         :raises NotFoundError: raised when the given task identifier
             is not found on the registry
         """
-        try:
-            self._rwlock.writer_acquire()
-            del self._tasks[task_id]
-        except KeyError:
+        self._rwlock.reader_acquire()
+        found = self.conn.exists(task_id)
+        self._rwlock.reader_release()
+
+        if not found:
             raise NotFoundError(element=str(task_id))
-        finally:
-            self._rwlock.writer_release()
+
+        self._rwlock.writer_acquire()
+        self.conn.expire(task_id, -1)
```

Comment:
`DEL` is reported as dangerous command: https://www.digitalocean.com/community/tutorials/how-to-install-and-secure-redis-on-debian-10#step-5-%E2%80%94-renaming-dangerous-commands

Reply:
Yeah I read it  but the rationale is it is dangerous because someone can delete keys. If you don't abuse of that is fine. Morover, I don't see the point of using `expire` when `rq` uses `delete` too.

https://github.com/rq/rq/blob/master/rq/job.py#L594